ng/AzureCLIScriptGenerator.ts

```diff
@@ -0,0 +1,194 @@
+            let subscriptionCommandItems = new CliCommandItem(CliCommandType.Subscription, 'az account', subscriptionActions, [], false);
+            this.commandItems.push(subscriptionCommandItems); 
+
+            let subscriptionLocationsActions: Array<ICliCommandAction> = new Array<ICliCommandAction>();
+            subscriptionLocationsActions.push({ actionName: 'list-locations', parameters: [], applicableSubItems: ['*'] });
+            let subscriptionLocationsCommandItems = new CliCommandItem(CliCommandType.SubscriptionLocations, 'az account', subscriptionLocationsActions, [], false);
+            this.commandItems.push(subscriptionLocationsCommandItems);
+
+            let resourceGroupsActions: Array<ICliCommandAction> = new Array<ICliCommandAction>();
+            resourceGroupsActions.push({ actionName: 'create', parameters: ['location', 'name'], applicableSubItems: ['*'] });
+            resourceGroupsActions.push({ actionName: 'delete', parameters: ['name'], applicableSubItems: ['*'] });
+            resourceGroupsActions.push({ actionName: 'list', parameters: [], applicableSubItems: ['*'] });
+            let resourceGroupsCommandItems = new CliCommandItem(CliCommandType.ResourceGroups, 'az group', resourceGroupsActions, [], true);
+            this.commandItems.push(resourceGroupsCommandItems);
+            
+            let resourceGroupActions: Array<ICliCommandAction> = new Array<ICliCommandAction>();
+            resourceGroupActions.push({ actionName: 'show', parameters: [], applicableSubItems: ['*'] });
+            resourceGroupActions.push({ actionName: 'update', parameters: [], applicableSubItems: ['*'] });
+            resourceGroupActions.push({ actionName: 'export', parameters: [], applicableSubItems: ['*'] });
+            let resourceGroupCommandItems = new CliCommandItem(CliCommandType.ResourceGroup, 'az group', resourceGroupActions, [], false);
+            this.commandItems.push(resourceGroupCommandItems);
+        }
+    }
+
+    export class CliScriptGenerator {
+        private static commandItems: Array<CliCommandItem> = [];
+
+
+        constructor(private resolver: CliCommandTypeResolver) {
+            // Invoke the static constructor to do a one time init
+            CliCommandCollection.initCommandItems();
+        }
+
+        private getScriptFor(commandItem: CliCommandItem) : string {
+            var script = "";
+            for (let actionItem of commandItem.getActions()) {
+                if (actionItem.applicableSubItems.includes('*')) {
+                    script += `${commandItem.getCliCmdlet()} ${actionItem.actionName}`;
+                    for (let parameter of actionItem.parameters) {
+                        script += ` --${parameter} ${this.resolver.getParameter(parameter)}`;
+                    }
+                    script += "\n\n";
+                }                
+            }
+            for (let actionItem of commandItem.getActions()) {
```

Comment:
i see 'for..of' gets transpiled to a regular for loop in the js file. so no difference at runtime. i also verified it on edge and worked fine.